hyperon_das_atomdb/adapters/ram_only.py

```diff
@@ -437,3 +515,33 @@ def add_link(self, link_params: Dict[str, Any], toplevel: bool = True) -> Dict[s
 
     def reindex(self, pattern_index_templates: Optional[Dict[str, Dict[str, Any]]]):
         raise NotImplementedError()
+
+    def delete_atom(self, handle: str, **kwargs) -> None:
+        node = self.db.node.pop(handle, None)
+
+        if node:
+            atom_type = node['named_type']
+
+            named_type_exists = any(
```

Comment:
In order to perform this single check,  you are iterating on all nodes. I understand why you did it but you need to be very careful to not introduce this kind of inefficiency in the code as it may pass unnoticed. When it seems we are forced to introduce such type of performance penalty we need to discuss. In this case, for example, it's acceptable just to let go the unused types. The cost of having then is irrelevant and we can always have some type of periodical (or explicitly user-requested) cleanup tasks to clean up this kind of thing out of the AtomDB.

I've create a card to make sure we don't forget this (https://github.com/singnet/das-atom-db/issues/90).

So you can remove this check and the code below which actually delete the atom type.

Reply:
Yes, I understand and agree with you. We can implement a routine task to cleanup these `atom_types` in the future. I was aware of the performance and execution on all nodes, but since we are on `ramOnly` I thought the atoms couldn't be too big. Actually thought I'd talk to you before implementing type removal. I imagine you would mention this point

Reply:
OK. Understood. An additional comment, though: `ramOnly` is the adapter used by most of users in the MeTTa interpreter. It will be used to run the majority of the user's experiments. It's not just a buffer to store temporary info before moving it to a remote DAS server. So we must keep strict focus to make all operations as optimized as we can.